compiler/mir/mirpasses.nim

```diff
@@ -455,6 +488,16 @@ proc applyPasses*(body: var MirBody, prc: PSym, env: var MirEnv,
       # requires the extraction for cstring literals
       extractStringLiterals(body.code, env, c)
 
+  # instrument the body with profiler calls after all lowerings, but before
+  # optimization
+  if (sfPure notin prc.flags) and (optProfiler in prc.options) and
+     (target in {targetC, targetJs}):
+    # XXX: not enabled for the VM because compile-time and run-time cannot
```

Comment:
Could have the compiler pass system temporarily remove the flag from options and then restore it prior to the code generation pass.

Reply:
Hm, no, I haven't considered doing this, but it would work, and the implementation cost is small. I'll implement it.